src/app/treasure-hunt/operation-costs/operation-costs.component.ts

```diff
@@ -425,4 +448,463 @@ export class OperationCostsComponent implements OnInit {
     this.save();
   }
 
+  calculateElectricityUnitCosts(){
```

Comment:
Initial glance...  I think there are a few opportunities to make things more DRY and clean up a bit. Take a look at template driven forms/forms through out the app and come back to this to improve.

Things to think about:
- Don't need the scoped annual___ variables defined at the start of each method
- Instead of using numerous check() methods, we _should_ be able to pass in args from the template to a single method that sets each variable representing error state. 
- Instead of "the rest of the form", tell the user specifically which fields need to be valid. 
- Is it necessary to track the error state of each property of each utility type or is there a way to simplify this?
- Anyway to genericize calculate methods or generally pull out repetitive logic